frame/support/src/dispatch.rs

```diff
@@ -3496,111 +3496,177 @@ mod tests {
 #[allow(dead_code)]
 mod weight_tests {
 	use super::*;
-	use sp_core::{parameter_types, Get};
+	use sp_core::parameter_types;
+	use sp_runtime::{generic, traits::BlakeTwo256};
 	use sp_weights::RuntimeDbWeight;
 
-	pub trait Config: 'static {
-		type RuntimeOrigin;
-		type Balance;
-		type BlockNumber;
-		type DbWeight: Get<RuntimeDbWeight>;
-		type PalletInfo: crate::traits::PalletInfo;
-	}
+	pub use self::frame_system::{Call, Config, Pallet};
 
-	pub struct TraitImpl {}
+	#[crate::pallet]
+	pub mod frame_system {
```

Comment:
I'd call this `system` to separate this from the external `frame_system` crate.

Reply:
If I name the module as `system`, I will need to rename the `system` module as `frame_system` again, like:

```rust
pub use self::system as frame_system;

#[crate::pallet]
pub mod system {
        use super::{frame_system, frame_system::pallet_prelude::*};
        ...
}
```

Otherwise, it will emit a compile error. 

```
error[E0277]: the trait bound `T: frame_system::Config` is not satisfied
    --> frame/support/src/dispatch.rs:3505:2
     |
3505 |     #[crate::pallet]
     |     ^^^^^^^^^^^^^^^^ the trait `frame_system::Config` is not implemented for `T`
     |
     = note: this error originates in the attribute macro `crate::pallet` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
     |
3511 |         #[pallet::pallet + frame_system::Config]
     |                          ++++++++++++++++++++++

error[E0277]: the trait bound `T: frame_system::Config` is not satisfied
```

It seems that the pallet macro does a mandatory check on the naming of the lowest-level module.

Reply:
Yeah seems like the neme `frame_system` is hardcoded. 

I am surprised `#[pallet::disable_frame_system_supertrait_check]` doesn't solve it.

Reply:
The error still appears when renaming to `test_pallet`.  
Resolving now since it is just the name of a mocked test pallet.